### DISCUSSION TITLE
PIL reader axis swap

### DISCUSSION THREAD

**Discussion by aeye-slater** [Project-MONAI/MONAI#7837] [General]
I am struggling to understand the reason behind the axis swap for PIL reader that swaps axis 0 and 1.  

Granted I can swap it back-- reading up it has something to do with the other writers and backends.  However the same image from loaded from DICOM, saved as a JPG, reloaded gets the axis swap when displayed.  If the axis swap is needed I should get the same image orientation back.

I've read about compatibility with other backend readers, but my experience so far is that MONAI is the only one among PIL, OpenCV, 3D Slicer, that appears to be doing this and I'm not understanding why this is needed.  We're also struggling with orienting 3d masks and volumes so this extra axis swap had caused us some pain when loading .seg.nrrd files as well trying to keep track of the all the axis changes.  It could be the seg.nrrd files were setup wrong, but even a simple jpg img off google makes no sense why it is being loaded with axis 0 (X) and axis (Y) swapped.

**Comment by KumoLiu** [comment]
Hi @aeye-slater, thanks for your interest here.

We have an argument `reverse_indexing` in `PILReader` which is set to `True` by default to make the reader's output consistent with the other readers. You can set this option to `False` to use the PIL backend's original spatial axes convention.
https://github.com/Project-MONAI/MONAI/blob/08d572839ede5d087e49d4dfadba033e512528df/monai/data/image_reader.py#L1136

Hope it helps, thanks.
